### PR TITLE
Improve handling of missing/corrupted port file

### LINKED ISSUE #1968: Don't abort on port file missing if --port is used.
The `--port` option can be used to target suites (found with `cylc scan`) if the port file is wrong (e.g. multiple instances of the same suite started accidentally).  At the moment it doesn't work if the port file is actually missing, because commands check for that and abort.

### PR DESCRIPTION
**Pull request by matthewrmshin**
If `--port=PORT` is set in a client CLI, the value will be used if the
port file is missing. Missing/corrupted port file is only fatal for the
client if `--port=PORT` is not set.

Suite now health checks its port file (missing, corrupted or modified) at
the same time as it checks the existence of its suite run directory.

Suite run directory and port file health check interval now
configurable via `suite.rc`. (This is mainly to facilitate testing.)

Close #1968.

@hjoliver @arjclark please review and sanity check.

### CODE REVIEW

File: lib/cylc/cfgspec/suite.py [hunk 1]
```diff
@@ -218,6 +218,8 @@ def _coerce_parameter_list(value, keys, _):
         'force run mode': vdr(
             vtype='string', options=['live', 'dummy', 'simulation', '']),
         'abort if any task fails': vdr(vtype='boolean', default=False),
+        'health check interval': vdr(
+            vtype='interval_minutes', default=DurationFloat(600)),
```
Comment (arjclark) [inline lib/cylc/cfgspec/suite.py]: Could we have this configurable at a site level too?

Comment (matthewrmshin) [inline lib/cylc/cfgspec/suite.py]: Now globally configurable.

Comment (arjclark) [inline lib/cylc/cfgspec/suite.py]: Cool.

**Comment by arjclark** [comment]
Looks OK to me. No problems from the test-battery in my environment. Over to you @hjoliver

**Comment by matthewrmshin** [comment]
Branch rebased.

**Comment by matthewrmshin** [comment]
Branch re-based.

**Comment by hjoliver** [comment]
I'm not sure that "health check interval" is an appropriate name for such a restricted kind of check - doesn't it suggest a wider ranging set of checks on "suite health" (conceivably even things like are there any failed tasks present...)?

Also, the documentation should say what happens if the check fails, and it should be documented in the global config reference too.

Otherwise good!

**Comment by matthewrmshin** [comment]
Hi @hjoliver I can see 3 options for *health check interval*.

1. Either we leave it as-is, but ensure that we extend the functionality to include other checks.
2. Change it to something like *port file and run directory check interval*.
3. Make it into a section in `global.rc` with items such as 2. This should allow the section to be extended to other checks.

Any opinion?

**Comment by hjoliver** [comment]
On reflection, I'll go with option 1.  Any alternative names I can think of are a bit ungainly, and we can always convert to a section later if we find that more checks are needed.

**Pull request merged**